workers/workers.go

```diff
@@ -0,0 +1,199 @@
+package workers
+
+import (
+	"time"
+
+	"github.com/serverless/event-gateway-connector/connection"
+	"github.com/serverless/event-gateway-connector/utils"
+	"github.com/serverless/event-gateway-connector/watcher"
+	"go.uber.org/zap"
+)
+
+// worker is the internal representation of the worker process
+type worker struct {
+	id     uint
+	recv   chan *connection.Connection
+	errors chan<- workerError
+	close  chan<- uint
+	log    *zap.SugaredLogger
+	done   chan bool
+}
+
+// WorkerPool is the default struct for our worker pool, containing mostly private values
+// including the maximum workers eligible, current count of workers, etc.
+type WorkerPool struct {
+	maxWorkers uint
+	numWorkers uint
+	log        *zap.SugaredLogger
+	jobs       map[connection.ID]*job // map of job handlers assigned to each connection.ID
+	events     <-chan *watcher.Event
+	done       chan bool // signal channel to stop all worker processes
+	stack      *utils.Stack
+}
+
+// job is the interim struct to manage the specific worker for a give connectionID
+type job struct {
+	workers map[uint]*worker
+	conn    *connection.Connection
+}
+
+// define the map of workers to manage
+var workerMap = make(map[uint]*worker)
+
+// workerError for cases where the worker ends up failing for a specific reason
+type workerError struct {
+	id     uint
+	connID connection.ID
+	err    error
+}
+
+// NewPool will accept a few initializer variables in order to stand up the new worker
+// pool of goroutines. These workers will listen for *watcher.Events and handle the
+// internal *Connection to manage data.
+func NewPool(log *zap.SugaredLogger, maxWorkers uint, events <-chan *watcher.Event) (*WorkerPool, error) {
```

Comment:
If the package name would be `workerpool` this would be just `New`. It's kinda convention that logger is passed as the last parameter. yeah, it might not be obvious :)